pkg/controller/jenkins/configuration/base/reconcile.go

```diff
@@ -177,13 +177,19 @@ func (r *ReconcileJenkinsBaseConfiguration) ensureResourcesRequiredForJenkinsPod
 	}
 	r.logger.V(log.VDebug).Info("Extra role bindings are present")
 
-	if err := r.createService(metaObject, resources.GetJenkinsHTTPServiceName(r.Configuration.Jenkins), r.Configuration.Jenkins.Spec.Service); err != nil {
+	httpServiceName := resources.GetJenkinsHTTPServiceName(r.Configuration.Jenkins)
+	if err := r.createService(metaObject, httpServiceName, r.Configuration.Jenkins.Spec.Service); err != nil {
 		return err
 	}
 	r.logger.V(log.VDebug).Info("Jenkins HTTP Service is present")
 	if err := r.createService(metaObject, resources.GetJenkinsSlavesServiceName(r.Configuration.Jenkins), r.Configuration.Jenkins.Spec.SlaveService); err != nil {
 		return err
 	}
+	if resources.RouteAPIIsDiscoverable {
+		if err := r.createRoute(metaObject, httpServiceName, r.Configuration.Jenkins); err != nil {
+			return err
+		}
+	}
 	r.logger.V(log.VDebug).Info("Jenkins slave Service is present")
```

Comment:
move this to 188 line

Reply:
fixed

Reply:
not sure to understand this one, but should be fixed now